lib/widgets/end_drawer/end_drawer.dart

```diff
@@ -21,11 +22,16 @@ import 'package:provider/provider.dart';
 
 import 'junto_themes_page.dart';
 
-class JuntoDrawer extends StatelessWidget {
-  Future<void> _onLogOut(BuildContext context) async {
+class JuntoDrawer extends StatefulWidget {
+  @override
+  _JuntoDrawerState createState() => _JuntoDrawerState();
+}
+
+class _JuntoDrawerState extends State<JuntoDrawer> {
+  Future<void> _onLogOut() async {
     try {
-      await context.bloc<AuthBloc>().add(LogoutEvent());
-      Navigator.popUntil(context, (r) => r.isFirst);
+      context.bloc<AuthBloc>().add(LogoutEvent());
+      Navigator.of(context).pushReplacement(Welcome.route());
```

Comment:
I think this breaks login flow i.e. if you got to login or sign up it won't work

Reply:
Checked this, sign in, and sign up works fine. 

Going to the first route does not guarantee the user would be taken to the login screen since for some the first route would be the lotus screen.